packages/block-editor/src/components/inserter/index.js

```diff
@@ -102,10 +128,62 @@ class Inserter extends Component {
 
 export default compose( [
 	withSelect( ( select, { rootClientId } ) => {
-		const { hasInserterItems } = select( 'core/block-editor' );
+		const { hasInserterItems, hasOneAllowedItem } = select( 'core/block-editor' );
 
 		return {
 			hasItems: hasInserterItems( rootClientId ),
+			hasOneAllowedItem: hasOneAllowedItem( rootClientId ),
```

Comment:
If I remember properly, there are multiple reasons that might result in just one item being available in the inserter (in addition to the allowed blocks property). Do we want to change the behavior everytime the inserter only shows one block (rely on the result of getInserterItems) or only apply the behavior once the allowed blocks lenght is 1?

Reply:
yes, not only for the inner blocks but in all cases we should not show a choice of one possible thing, that's my opinion :)

Reply:
What was the verdict here?